09 Redux/src/reducers/memberReducer.ts

```diff
@@ -0,0 +1,83 @@
+import MemberEntity from "../api/memberEntity";
+import MemberAPI from "../api/memberAPI";
+import objectAssign = require('object-assign');
+import MemberFormErrors from "../validations/memberFormErrors"
+import MemberFormValidator from "../validations/memberFormValidator"
+
+
+let emptyMemberEntity = new MemberEntity();
+
+class MemberState  {
+  member : MemberEntity;
+  memberId : number;
+  errors : MemberFormErrors;
+  isValid : boolean;
+  saveCompleted : boolean;
+
+  public constructor()
+  {
+    this.member = new MemberEntity();
+    this.memberId = -1;
+    this.errors = new MemberFormErrors();
+    this.isValid = false;
+    this.saveCompleted = false;
+  }
+}
+
+
+// Just to show how combine reducers work, we have
+// divided into two reducers member load + member load/update/delete
+export default (state : MemberState = new MemberState(), action) => {
+  let newState : MemberState = null;
+
+  switch (action.type) {
+    case 'MEMBER_NEW':
+      newState = objectAssign({}, state, {member: new MemberEntity(), errors: new MemberFormErrors(), isValid: false});
+      return newState;
+
+    case 'MEMBER_LOAD':
+      let member : MemberEntity;
+      let memberId : number = action["id"];
+
+      member = MemberAPI.getMemberById(memberId);
+      newState = objectAssign({}, state, {dirty: false, member: member, errors: new MemberFormErrors(), isValid: true});
+
+      return newState;
+
+    case 'MEMBER_UI_INPUT':
```

Comment:
In this case I will keep it as it is: I add the "MEMBER" prefix to classify const and avoid colllisions with other reducers, and UI input means something came from input (maybe it would be more accurate MEMBER_UPDATE_UI_INPUT_SINGLE_VALUE, but maybe that's too verbose for this small sample?)